File: DRST manager/EditViewController.m
//
//  EditViewController.m
//  DRST manager
//
//  Created by 김승호 on 2015. 11. 21..
//  Copyright © 2015년 Seungho Kim. All rights reserved.
//

#import "EditViewController.h"

@interface EditViewController ()

@property DataKit *dk;
@property int deviceIndex;
@property int max;
@property int cur;

@end

@implementation EditViewController

- (void)viewDidLoad {
	[super viewDidLoad];
	_dk = [[DataKit alloc] init];
	_deviceIndex = 0;
	_segmentedButton.hidden = ![self.dk dualAccountEnabled];
	[self load];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (IBAction)txtMaxTouched:(id)sender {
	[self setMaximumValue:(int)[_txtMax.text integerValue]];
}

- (IBAction)TextCurrentTouched:(id)sender {
	[self setCurrentValue:(int)[_txtCurrent.text integerValue]];
}

- (IBAction)stepMaxTouched:(id)sender {
	[self setMaximumValue:_stepMax.value];
}

- (IBAction)stepCurrentTouched:(id)sender {
	[self setCurrentValue:_stepCurrent.value];
}

- (void)save {
	[self.view endEditing:YES];
	[self.dk setMaxStamina:_max atIndex:_deviceIndex];
	[self.dk setCurrentStamina:_cur atIndex:_deviceIndex];
	[self.dk setDate:[NSDate date]];
	if ([self.dk notificationEnabled]) {
		[NotificationKit registerNotification];
	} else {
		[NotificationKit clearNotification];
	}
}

- (IBAction)save:(id)sender {
	[self save];
	[self.navigationController popViewControllerAnimated:YES];
}

- (IBAction)segmentedButtonTouched:(id)sender {
	[self save];
	_deviceIndex = (int)_segmentedButton.selectedSegmentIndex;
	[self load];
}

- (void)load {
	[self setMaximumValue:(int)[self.dk maxStamina:_deviceIndex]];
	[self setCurrentValue:(int)[self.dk estimatedCurrentStamina:_deviceIndex]];
}

- (void)setMaximumValue:(int)value {
	if (value > _stepMax.maximumValue) {
		value = _stepMax.maximumValue;
	} else if (value < _stepMax.minimumValue) {
		value = _stepMax.minimumValue;
	}
	_max = value;
	_txtMax.text = [NSString stringWithFormat:@"%d", value];
	_stepMax.value = value;
	_stepCurrent.maximumValue = value;
}

- (void)setCurrentValue:(int)value{
	if (value > _stepCurrent.maximumValue) {
		value = _stepCurrent.maximumValue;
	} else if (value < _stepCurrent.minimumValue) {
		value = _stepCurrent.minimumValue;
	}
	_cur = value;
	_txtCurrent.text = [NSString stringWithFormat:@"%d", value];
	_stepCurrent.value = value;
	_stepMax.minimumValue = MAX(value, 40);
}

- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event
{
	[self.view endEditing:YES];
}

@end

File: DRST manager/EditViewController.swift
//
//  EditViewController.swift
//  DRST manager
//
//  Created by 김승호 on 2016. 1. 30..
//  Copyright © 2016년 Seungho Kim. All rights reserved.
//

import UIKit
import DRSTKit

class EditViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
	var deviceIndex = 0
	var stamina = (max:40, current:0)
	let dk = DataKit.init()
	
	@IBOutlet weak var maxText: UITextField!
	@IBOutlet weak var currentText: UITextField!
	@IBOutlet weak var maxStepper: UIStepper!
	@IBOutlet weak var currentStepper: UIStepper!
	@IBOutlet weak var segmentedButton: UISegmentedControl!
	
	override func viewDidLoad() {
		super.viewDidLoad()
		deviceIndex = 0
		segmentedButton.isHidden = !dk.dualAccountEnabled
		load()
	}
	
	func load() {
		setMaximumValue(dk.maxStamina(deviceIndex))
		setCurrentValue(dk.estimatedCurrentStamina(deviceIndex))
	}
	
	func save() {
		view.endEditing(true)
		dk.setMaxStamina(stamina.max, atIndex: deviceIndex)
		dk.setCurrentStamina(stamina.current, atIndex: deviceIndex)
		dk.date = Date.init()
		if dk.notificationEnabled {
			DRSTNotification.register()
		} else {
			DRSTNotification.clear()
		}
	}
	
	override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
		view.endEditing(true)
	}
	
	func setMaximumValue(_ value:Int) {
		stamina.max = max(min(value, Int(maxStepper.maximumValue)), Int(currentStepper.minimumValue))
		maxText.text = String(stamina.max)
		maxStepper.value = Double(stamina.max)
		currentStepper.maximumValue = Double(stamina.max)
	}
	
	func setCurrentValue(_ value:Int) {
		stamina.current = max(min(value, Int(currentStepper.maximumValue)), Int(currentStepper.minimumValue))
		currentText.text = String(stamina.current)
		currentStepper.value = Double(stamina.current)
		maxStepper.minimumValue = Double(max(stamina.current, 40))
	}
	
	@IBAction func textMaxTouched(_ sender: AnyObject) {
		if (maxText.text == "") {
			maxText.text = String(stamina.max)
		} else {
			setMaximumValue(Int(maxText.text!)!)
		}
	}
	
	@IBAction func textCurrentTouched(_ sender: AnyObject) {
		if (currentText.text == "") {
			currentText.text = String(stamina.current)
		} else {
			setCurrentValue(Int(currentText.text!)!)
		}
	}
	
	@IBAction func maxStepperTouched(_ sender: AnyObject) {
		setMaximumValue(Int(maxStepper.value))
	}
	
	@IBAction func currentStepperTouched(_ sender: AnyObject) {
		setCurrentValue(Int(currentStepper.value))
	}
	
	@IBAction func save(_ sender: AnyObject) {
		save()
		deviceIndex = segmentedButton.selectedSegmentIndex
		load()
		_ = navigationController?.popViewController(animated: true)
	}
	
	@IBAction func segmentedButtonTouched(_ sender: AnyObject) {
		save()
		deviceIndex = segmentedButton.selectedSegmentIndex
		load()
	}
}
